addons/binding/org.openhab.binding.zoneminder/src/main/java/org/openhab/binding/zoneminder/handler/ZoneMinderServerBridgeHandler.java

```diff
@@ -618,276 +606,692 @@ private void onDisconnected() {
+                statusDetail = ThingStatusDetail.CONFIGURATION_ERROR;
+                statusDescription = "Telnet port invalid in configuration";
+                updateStatus(newStatus, statusDetail, statusDescription);
+                logger.error("{}: context='{}' state='{}' check='FAILED' - {}", getLogIdentifier(), context,
+                        newStatus.toString(), statusDescription);
+                return status;
+
+            }
+
+            // Configuration verified
+            status = ZoneMinderConnectionStatus.BINDING_CONFIG_LOAD_PASSED;
+
+            logger.debug("{}: context='{}' state='{}' check='PASSED'", getLogIdentifier(), context,
+                    newStatus.toString());
+
+        } catch (Exception ex) {
+            logger.error("{}: context='{}' state='{}' check='FAILED' tag='exception'", getLogIdentifier(), context,
+                    newStatus.toString(), ex);
+
+        }
+        return status;
+    }
+
+    private ZoneMinderConnectionStatus validateConfig(ZoneMinderBridgeServerConfig config) {
+        String context = "validateConfig";
+        ThingStatus newStatus = ThingStatus.OFFLINE;
+        ThingStatusDetail statusDetail = ThingStatusDetail.NONE;
+        String statusDescription = "";
+
+        ZoneMinderConnectionStatus status = ZoneMinderConnectionStatus.GENERAL_ERROR;
+
+        // Is it a retry loop? (Is this step already verified?)
+        // Or is there an unrecoverable error?
+        if ((getLastSucceededStatus().greatherThanEqual(ZoneMinderConnectionStatus.BINDING_CONFIG_VALIDATE_PASSED))
+                || (getLastSucceededStatus().hasUnrecoverableError())) {
+            return getLastSucceededStatus();
+        }
+
+        // Check if something is responding give host and port HTTP
         try {
-            // Just perform a health check to see if we are still connected
-            if (prevStatus == ThingStatus.ONLINE) {
-                if (zoneMinderSession == null) {
-                    newStatus = ThingStatus.ONLINE;
-                    statusDetail = ThingStatusDetail.NONE;
-                    statusDescription = "";
-                    updateBridgeStatus(newStatus, statusDetail, statusDescription);
-                    return;
-                } else if (!zoneMinderSession.isConnected()) {
-                    newStatus = ThingStatus.OFFLINE;
-                    statusDetail = ThingStatusDetail.COMMUNICATION_ERROR;
-                    statusDescription = "Session lost connection to ZoneMinder Server";
-                    updateBridgeStatus(newStatus, statusDetail, statusDescription);
+            // Check Socket
+            Socket socket = new Socket();
+            socket.connect(new InetSocketAddress(config.getHost(), config.getHttpPort()), 5000);
+            if (socket.isConnected()) {
+                socket.close();
+            }
 
-                    return;
-                }
+            newStatus = ThingStatus.OFFLINE;
```

Comment:
Why OFFLINE, you are even connecting, maybe UNKNOW would be more suitable?

Reply:
Nope, at that specific place I have verified that config in openHAB looks good, Now I have to verify the remote configuration, which could also be wrong. The only way to find out is trial and error . I will add comments in the code to explain :-)